response.cc

```diff
@@ -54,29 +54,12 @@ std::unique_ptr<Response> Response::Parse(const std::string& raw_res)
 void Response::SetStatus(const ResponseCode response_code)
 {
 	response_status_ = response_code;
-	switch(response_code)
-	{
-	case Response::ok:
-		response_status_first_line_ = status_string::ok;
-		break;
-	case Response::moved_permanently:
-		response_status_first_line_ = status_string::moved_permanently;
-		break;
-	case Response::found:
-		response_status_first_line_ = status_string::found;
-		break;
-	case Response::bad_request:
-		response_status_string_ = status_string::bad_request;
-		break;
-	case Response::not_found:
-		response_status_first_line_ = status_string::not_found;
-		break;
-	case Response::internal_server_error:
-		response_status_first_line_ = status_string::internal_server_error;
-		break;
-	default:
-		response_status_first_line_ = status_string::internal_server_error;
-		break;
+;	try {
+		response_status_first_line_ = status_strings.at(response_code);
+	}
+	catch (...) {
+		BOOST_LOG_TRIVIAL(error) << "Status code " << response_code << " has no corresponding status_string (SetStatus)";
+		response_status_first_line_ = "";
```

Comment:
What happens here if the response code doesn't exist? The first line is empty so it would create an invalid response? Maybe have it be default to 404 error or something?